addons/l10n_id/models/account_move.py

```diff
@@ -0,0 +1,152 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import pytz
+
+from odoo import api, fields, models, _
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    # Used to fetch the QR payment status.
+    l10n_id_qris_invoice_details = fields.Json(
+        string="QRIS Transaction Number",
+        help="Transaction Number stored.",
+        readonly=True,
+        export_string_translation=False,
+    )
+    # Used to return the same QR code if still valid.
+    l10n_id_latest_qris_content = fields.Char(
+        string="QRIS Content",
+        help="QRIS Content stored as a char.",
+        readonly=True,
+        export_string_translation=False,
+    )
+
+    def _generate_qr_code(self, silent_errors=False):
+        """
+        Adds information about which invoice is triggering the creation of the QR-Code, so that we can link both together.
+        """
+        self.env.context = {**self.env.context, "qris_originating_invoice_id": self.id}
```

Comment:
Never modify the context like that :scream: 
Use `super(AccountMove, self.with_context(...))._generate_qr_code` instead